wordpress-mobile/WordPress-Android
# Disable local notifications with Notification Settings switch

Fixes #15816 

This prevents local notifications when the Notification Settings switch is off. Prevented local notifications are
- Weekly roundup notifications,
- Blogging reminder notifications,
- Create site notifications.

Blogging reminders popups are still active and can be displayed in some cases (eg. from Site Settings) even Notification Settings switch is off. If the user sets a new reminder from the popup, it will be scheduled even if the switch is off.

**Merge requirements:**
- [x] 1. Merge [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2571)
- [x] 2. Update FluxC reference to `trunk`
- [x] 3. Remove `[Status] Not Ready for Merge` label
- [x] 4. Merge this

### To test:
You can wait for scheduled dates to test notifications. It would be the best method but testing all cases takes an unrealistic long time. Alternatively, you can check [App Inspection > Background Task Inspector](https://developer.android.com/studio/inspect/task#inspect-jobs-alarms-wakelocks) on Android Studio if these notifications are scheduled.

**Weekly roundup notifications**
1. Launch the WordPress app. 
2. Navigate to Notifications.
3. Tap ⚙️ button at the top of the screen.
4. Turn off the switch at the top of the page.
5. Verify `WeeklyRoundupWorker` is canceled on `App Inspection > Background Task Inspector`.
6. Turn on the switch again. 
7. Verify `WeeklyRoundupWorker` is scheduled on `App Inspection > Background Task Inspector`.
8. Repeat 1-7 for Jetpack app.

**Blogging reminder notifications**
1. Launch the WordPress app. 
2. Navigate to `My Site → MENU → Site Settings`.
3. Tap Blogging Reminders.
4. Set a reminder.
5. Verify `ReminderWorker` is scheduled on `App Inspection > Background Task Inspector`.
6. Navigate back and change your selected site from the site picker at the top of the MENU page.
7. Navigate to Site Settings and set another reminder for the selected site.
8. Verify a new `ReminderWorker` is scheduled on `App Inspection > Background Task Inspector`.
9. Navigate to Notifications. Tap ⚙️ button at the top of the screen.
10. Turn off the switch at the top of the page.
11. Verify both `ReminderWorker`s are canceled on `App Inspection > Background Task Inspector`.
12. Turn on the switch.
13. Verify both `ReminderWorker`s are scheduled on `App Inspection > Background Task Inspector`.
14. Repeat 1-13 for Jetpack app.

**Create site notifications**
1. Uninstall the app if it's installed.
2. Install and launch the WordPress app. 
3. Log in with an account without any site.
4. Verify `LocalNotificationWorker` is scheduled on `App Inspection > Background Task Inspector`.
5. Navigate to Notifications.
6. Tap ⚙️ button at the top of the screen.
7. Turn off the switch at the top of the page.
8. Verify `LocalNotificationWorker` is canceled on `App Inspection > Background Task Inspector`.
**NOTE:** Create site notifications won't be rescheduled if you turn on the switch again. Two Create site notifications are scheduled when the app is installed the first time. One is for the next day after the install, another one is for 8 days after the first install. If the switch is turned off and turned on in 8 days, create site notifications won't display. I'll discuss this later and change it in a separate PR if necessary.

## Regression Notes
1. Potential unintended areas of impact
Rescheduling blogging reminders after turning Notification Settings off and on again.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually.

3. What automated tests I added (or what prevented me from doing so)
Update `CreateSiteNotificationHandlerTest`.

PR submission checklist:
